packages/cli-kit/src/node/cli.ts

```diff
@@ -52,4 +53,36 @@ export async function runCreateCLI(options: RunCLIOptions) {
   await runCLI(options)
 }
 
+export async function replaceGlobalCLIWithLocal(filepath: string): Promise<boolean> {
+  const cliPackage = await localCliPackage()
+  if (!cliPackage) return false
+
+  const correctExecutablePath = join(cliPackage.path, cliPackage.bin.shopify)
+  if (correctExecutablePath === filepath) return false
+  try {
+    await exec(correctExecutablePath, process.argv.slice(2, process.argv.length), {stdio: 'inherit'})
```

Comment:
We should also `debug` if we hit this path, probably including printing out the args and the path.